Update g-w to cycle with C1152 ATM

CatherineThomas-NOAA opened the pull request:
# Description

There are a few updates needed to cycle with the C1152 atmospheric model:
- config.resources(.$machine): include C1152 case options for DA tasks
- gdasfcst config.ufs: WRTTASK_PER_GROUP_PER_THREAD_PER_TILE_GDAS=20 
- source code update for calc_analysis (https://github.com/NOAA-EMC/GSI-utils/pull/59) for memory
- resource change needed for atmanl upp

Note that any needed changes to time limits for C1152 will be addressed at a later date.

Resolves #3173 


# Type of change
- [x] New feature (adds functionality)

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? YES
- Does this change require an update to any of the following submodules? YES
  - [x] GSI-utils NOAA-EMC/GSI-Utils#59

# How has this been tested?
- Cycled C1152/C384 ATM-only on WCOSS2

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

CatherineThomas-NOAA committed:
Updates for cycling C1152 ATM

A handful of updates are needed to cycle with C1152
atmosphere, which are mostly related to configs and
resources.

Refs: #3173

CatherineThomas-NOAA committed:
Merge branch 'NOAA-EMC:develop' into c1152_cyc

CatherineThomas-NOAA committed:
Merge branch 'NOAA-EMC:develop' into c1152_cyc

CatherineThomas-NOAA committed:
Update gsi_utils hash for C1152 cycling

The calc_analysis utility was updated to handle the
increased horizontal resolution of C1152 ATM.

Refs: #3173

aerorahul reviewed (dismissed):
looks good to me.

aerorahul commented:
Running the CI on Hera and Hercules to cover the gsi-utils hash update.  The other changes will not be tested in the CI

RussTreadon-NOAA commented:
The gfs and gdas analcalc failures in C96C48_ufs_hybatmDA are true errors.  

`calc_anl.x` aborts with 
```
 Computing and Adding Surface Pressure Increment
 rank of data array != variable ndims (or ndims-1)
...
 rank of data array != variable ndims (or ndims-1)
99
srun: Received task exit notification for 31 tasks of StepId=4860606.2 (status=0x6300).
srun: error: h6c28: tasks 96-126: Exited with exit code 99

```
when processing ensemble resolution files.

RussTreadon-NOAA commented:
Examine the guess and increment files read by the calc_analysis code.   GSI based atmospheric increment files contain
```
float delp_inc(lev, lat, lon) ;
```
whereas GDASApp based atmospheric increment files contain
```
        double delp_inc(time, lev, lat, lon) ;
```
GDASApp increment files add a time dimension to the increment array.

RussTreadon-NOAA commented:
@CatherineThomas-NOAA , the `sorc/gsi_utils.fd` hash in `CatherineThomas-NOAA:c1152_cyc` can be updated to [`f716012`](https://github.com/NOAA-EMC/GSI-utils/commit/f716012812c2564e7eab24041f7a3ec14c7aa383).  

The changes in [`f716012`](https://github.com/NOAA-EMC/GSI-utils/commit/f716012812c2564e7eab24041f7a3ec14c7aa383) have been tested in g-w CI C96C48_hybatmDA and C96C48_ufs_hybatmDA on Cactus.  gfs_analcalc and gdas_analcalc successfully ran to completion in both configurations.

CatherineThomas-NOAA committed:
Update gsi_utils for C1152 cycling

The original calc_analysis bugfix was not suitable for
JEDI-based ATM DA.  This hash update allows for
calc_analysis to work with both GSI and JEDI-based ATM DA.

Resolves #3173

CatherineThomas-NOAA committed:
Merge branch 'NOAA-EMC:develop' into c1152_cyc

CatherineThomas-NOAA commented:
Thanks @RussTreadon-NOAA for the gsi_utils update. 

I tested the gsi_utils update in my C1152 exp and had no issues.  I've updated my fork with the new gsi_utils hash as well as merged in the latest g-w develop.  I think this PR is ready to retest @aerorahul.

WalterKolczynski-NOAA commented:
This passed on Hera but the label didn't get set.
